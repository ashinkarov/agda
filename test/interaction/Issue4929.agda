
postulate
  ANY : ∀{a}{A : Set a} → A

data Ty : Set where
  _⇒_ : (a b : Ty) → Ty

data Tm : (b : Ty) → Set where
  S   : ∀{c a b} → Tm ((c ⇒ (a ⇒ b)) ⇒ ((c ⇒ a) ⇒ (c ⇒ b)))
  _∙_ : ∀{a b} (t : Tm (a ⇒ b)) (u : Tm a) → Tm b

data _↦_ : ∀{a} (t t' : Tm a) → Set where

  ↦S : ∀{c a b} {t : Tm (c ⇒ (a ⇒ b))} {u : Tm (c ⇒ a)} {v : Tm c}
     → (((S ∙ t) ∙ u) ∙ v) ↦ ((t ∙ v) ∙ (u ∙ v))

  ↦r : ∀{a b} {u u' : Tm a} {t : Tm (a ⇒ b)}
     → (t ∙ u) ↦ (t ∙ u')

data SN {a} (t : Tm a) : Set where
  acc : (h : ∀{t'} (r : t ↦ t') → SN t') → SN t

test : ∀{a b} {t : Tm (a ⇒ b)} {u : Tm a} → SN t → SN u → SN (t ∙ u)
test snt snu = acc λ{ (↦S) → ANY ; (↦r) → {!test snt ?!} }

  -- Refine with (test snt)
  -- or give (test snt ?)

-- An internal error has occurred. Please report this as a bug.
-- Location of the error: src/full/Agda/TypeChecking/Substitute.hs:456
