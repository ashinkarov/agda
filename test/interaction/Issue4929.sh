#!/bin/sh

AGDA=$1
DIR=$PWD

$AGDA --interaction --ignore-interfaces <<EOF
IOTCM "$DIR/Issue4929.agda" NonInteractive Indirect ( Cmd_load "$DIR/Issue4929.agda" [] )
IOTCM "$DIR/Issue4929.agda" NonInteractive Indirect ( Cmd_give WithoutForce 0 (intervalsToRange (Just (mkAbsolute "$DIR/Issue4929.agda")) [Interval (Pn () 657 24 45) (Pn () 667 24 55)]) "test snt ?" )
EOF
