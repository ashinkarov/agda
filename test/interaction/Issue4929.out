(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue4929 (Issue4929.agda). " t)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : SN (t ∙ u') " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "An internal error has occurred. Please report this as a bug. Location of the error: __IMPOSSIBLE__, called at src/full/Agda/TypeChecking/Substitute.hs:«line»:«col» «Agda-package»:Agda.TypeChecking.Substitute " nil)
(agda2-highlight-add-annotations 'nil)
(agda2-status-action "")
